feat: Use picture el for poster

mister-ben opened the pull request:
## Description
Use a picture element instead of background image for the poster. Resolves issues some have had with the PosterImage and video el loading the poster differently.

Leaves scope to add support for source set for multiple image formats later, which could be done as a backwards-compatible minor.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chrome, Firefox, IE)
  - [x] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

gkatsev commented on src/js/poster-image.js:
```diff
@@ -46,12 +46,16 @@ class PosterImage extends ClickableComponent {
    */
   createEl() {
-    const el = Dom.createEl('div', {
+    const crossOrigin = this.player_.options_.crossOrigin || this.player_.options_.crossorigin || null;
```
Should we use the `crossOrigin` getter?
```suggestion
    const crossOrigin = this.player_.crossOrigin() ?? null;
```

mister-ben replied:
The poster can be set up before the getter will return a correct value, as it's dependent on the tech

gkatsev replied:
Ah, that's right. However, we should probably still use the getter first because otherwise, the value may be out of sync with the rest of the system.

gkatsev commented on src/js/player.js:
```diff
@@ -839,6 +839,9 @@ class Player extends Component {
     }
 
     this.techCall_('setCrossOrigin', value);
+    if (this.posterImage) {
+      this.posterImage.$('img').crossOrigin = value;
+    }
```
thought: maybe PosterImage should have a crossOrigin getter/setter so that this piece can be moved into it?
```
if (this.posterImage) {
  this.posterImage.crossOrigin(value);
}
```
If we do, we'll probably want to update the createEl below to use this option.

mister-ben commented:
Updated to add a getter/setter to posterImage. The getter gets posterImage's value if the el has been created, otherwise the player/tech's vlaue if the tech is ready, otherwise from the player options.
Updated the player setter to accept null as a value, which is needed to unset the value.
